### PR TITLE
Dashboard: fail build when webpack has errors

### PR DESCRIPTION
**Pull request by ljelinkova**
### What does this PR do?
Changes the dashboard build so that when typescript compilation fails (or anything processed by webpack), the whole build fails too. The watch mode should not fail in case of webpack error. 

Note that this PR depends on resolving all current compilation problems (e.g https://github.com/eclipse/che/pull/12826)
### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/12902

### CODE REVIEW

**Review by ashumilova — CHANGES_REQUESTED** [review]
The introduced changes make sense, but IDE2 teams needs to fix compilation errors before merging this PR.

**Comment by ashumilova** [comment]
PR from IDE2 team https://github.com/eclipse/che/pull/12955

**Comment by musienko-maxim** [comment]
Can one of the admins verify this PR?

**Comment by ashumilova** [comment]
@ljelinkova checked on my side - it works. But need to update the branch with master to pass checks on this PR.

**Comment by ljelinkova** [comment]
I've rebased to master.

**Comment by ashumilova** [comment]
ci-build

**Comment by benoitf** [comment]
ci-build

File: dashboard/gulp/scripts.js
Comment (olexii4) [inline dashboard/gulp/scripts.js]: We already have implementation for an error handler:
``` conf.errorHandler('Webpack')(err);```. 
It is better to revert the old implementation and improve exports.errorHandler
https://github.com/eclipse/che/pull/12911/files#diff-b7e557f6e1ded3f027a85e6b93a04243L85
or rework this line with using the same format
 ``` gutil.log(gutil.colors.red('[' + title + ']'), err.toString());  ```

File: dashboard/gulp/scripts.js
Comment (olexii4) [inline dashboard/gulp/scripts.js]: ``` if (watch && !callbackCalled) ```
I think it is better

Comment (ljelinkova) [reply dashboard/gulp/scripts.js]: OK

File: dashboard/gulp/scripts.js
Comment (ljelinkova) [reply dashboard/gulp/scripts.js]: I couldn't use `conf.errorHandler('Webpack')(err);.` because there is no err object available. I will change the log to  `$.util.log($.util.colors.red('[Webpack task failed with errors]'));` so that the text will be red.

**Review by olexii4 — APPROVED** [review]

**Comment by ljelinkova** [comment]
@ashumilova  @olexii4  What blocks this PR from merging?

**Comment by benoitf** [comment]
ci-build

**Comment by benoitf** [comment]
ci-test

**Comment by benoitf** [comment]
@ljelinkova there are webpack errors thus the build is failing

**Comment by olexii4** [comment]
ci-test

**Review by ashumilova — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

**Comment by olexii4** [comment]
crw-ci-test

**Comment by olexii4** [comment]
crw-ci-test

**Comment by olexii4** [comment]
ci-test

**Comment by olexii4** [comment]
crw-ci-test

**Comment by olexii4** [comment]
@ljelinkova   It works. But need to update the branch with master to pass checks on this PR.

**Comment by olexii4** [comment]
ci-test

**Comment by olexii4** [comment]
ci-build

**Pull request merged**
